src/main/java/org/openrewrite/jenkins/github/AddSecurityScanWorkflow.java

```diff
@@ -0,0 +1,171 @@
+    private static final String DEFAULT_WORKFLOW_PATH = "/org/openrewrite/jenkins/github/jenkins-security-scan.yml";
+    private static final JsonPathMatcher BRANCHES_KEY = new JsonPathMatcher("$.on.push.branches");
+    private static final String JAVA_VERSION_KEY = "jobs.security-scan.with.java-version";
+    private static final String BUILD_TOOL_KEY = "jobs.security-scan.with.java-cache";
+
+    @Option(displayName = "Branches",
+            description = "Run workflow on push to these branches.",
+            example = "main",
+            required = false)
+    @Nullable
+    List<String> branches;
+
+    @Option(displayName = "Java Version",
+            description = "Version of Java to set for build.",
+            example = "11",
+            required = false)
+    @Nullable
+    Integer javaVersion;
+
+    @Option(displayName = "Build Tool",
+            description = "Set up dependency cache.",
+            example = "maven",
+            valid = {"maven", "gradle"},
+            required = false)
+    @Nullable
+    String buildTool;
+
+    @Override
+    public String getDisplayName() {
+        return "Add Jenkins Security Scan workflow";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Adds the Jenkins Security Scan GitHub Actions workflow. " +
+                "See [docs](https://www.jenkins.io/doc/developer/security/scan/) for details.";
+    }
+
+    @Override
+    public AtomicBoolean getInitialValue(ExecutionContext ctx) {
+        return new AtomicBoolean();
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getScanner(AtomicBoolean found) {
+        Path path = Paths.get(FILE_PATH);
+        return new TreeVisitor<Tree, ExecutionContext>() {
+            @Override
+            public Tree visit(@Nullable Tree tree, ExecutionContext executionContext, Cursor parent) {
+                SourceFile sourceFile = (SourceFile) requireNonNull(tree);
+                if (path.toString().equals(sourceFile.getSourcePath().toString())) {
+                    found.set(true);
+                }
+                return sourceFile;
+            }
+        };
+    }
+
+    @Override
+    public Collection<? extends SourceFile> generate(AtomicBoolean found, ExecutionContext ctx) {
+        if (found.get()) {
+            return Collections.emptyList();
+        }
+        YamlParser parser = new YamlParser();
+        @Language("yml")
+        String workflow = defaultWorkflow();
+        return parser.parse(workflow)
+                .map(brandNewFile -> (Yaml.Documents) brandNewFile.withSourcePath(Paths.get(FILE_PATH)))
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor(AtomicBoolean found) {
+        return new YamlIsoVisitor<ExecutionContext>() {
+            @Override
+            public Yaml.Documents visitDocuments(Yaml.Documents documents, ExecutionContext executionContext) {
```

Comment:
I like how you're splitting the work up into separate visitors in this method, mostly reusing existing recipes where you can.

Reply:
Thanks, great to know this is how to do it!